Data Preparation.R

```diff
@@ -0,0 +1,173 @@
+library(tidyverse)
+library(janitor)   
+library(tictoc)
+
+
+## 2. Define dates ##
+
+extract_date <- "_extracted_22072019"
+
+start_date        <- "2009-10-01"  # start date for SMR01 and SMR50 extracts
+end_date          <- "2018-03-31"  # end date for SMR01 and SMR50 extracts
+deaths_start_date <- "2010-04-01"  # start date for deaths extract
+
+## 3. Set filepaths ##
+
+# Output filepath
+output <- "/PHI_conf/irf/EOL_Publication/Data preparation/RAP/Data Preparation/"
+
+# Lookup filepaths
+pc_lookup   <- "/conf/linkage/output/lookups/Unicode/Geography/
+                Scottish Postcode Directory/
+                Scottish_Postcode_Directory_2019_1.5.sav"
+simd_lookup <- "/conf/linkage/output/lookups/Unicode/Deprivation/
+                postcode_2019_1.5_simd2016.sav"
+locality    <- "/isdsf00d03/cl-out/lookups/Unicode/Geography/HSCP Locality/
+                HSCP Localities_DZ11_Lookup_20180903.sav"
+
+# SMRA connection
+SMRA <- suppressWarnings(dbConnect(odbc(), dsn = "SMRA",
+                                   uid = .rs.askForPassword("SMRA Username:"),
+                                   pwd = .rs.askForPassword("SMRA Password:")))
+
+## 4. Define external causes of death ##
+
+external <- c(paste0("V", 0, 1:9), paste0("V", 10:99),
+              paste0("W", 20:99),
+              paste0("X", 0, 0:9), paste0("X", 10:99),
+              paste0("Y", 0, 0:9), paste0("Y", 10:84))
+
+
+
+######################################
+## SECTION 2 - Data extraction      ##
+######################################
+
+
+
+## 1. SMR01 extract ##
+Query_SMR01 <- paste0("select link_no, gls_cis_marker, admission_date, 
+                       discharge_date, sex, ",
+                      "inpatient_daycase_identifier from SMR01_PI ",
+                      "where discharge_date >= {d '", start_date, "'} 
+                       AND discharge_date <= {d '", end_date, "'} ",
+                      "ORDER BY link_no, cis_marker, admission_date, 
+                       discharge_date, admission, discharge, uri")
+
+data_01 <- as_tibble(dbGetQuery(SMRA, Query_SMR01)) %>%
+           mutate(recid = "SMR01") %>% clean_names()
+
+
+## 2. SMR50 extract ##
+Query_SMR50 <- paste0("select link_no, gls_cis_marker, admission_date, 
+                       discharge_date, sex, ",
+                      "inpatient_daycase_identifier from ANALYSIS.SMR01_1E_PI ",
+                      "where discharge_date >= {d '", start_date, "'} 
+                       AND discharge_date <= {d '", end_date, "'} ",
+                      "ORDER BY link_no, cis_marker, admission_date, 
+                       discharge_date, admission, discharge, uri")
+
+data_50  <- as_tibble(dbGetQuery(SMRA, Query_SMR50)) %>%
+            mutate(recid = "SMR50") %>% clean_names()
+
+
+## 3. SMR04 extract ##
+Query_SMR04 <- paste("select link_no, cis_marker, admission_date, 
+                      discharge_date, sex,",
+                     "management_of_patient from ANALYSIS.SMR04_PI ",
```

Comment:
Don't need to extract management_of_patient as this is already filtered in the SQL query.